engine.php

```diff
@@ -1,70 +1,42 @@
 <?php
 
+declare(ticks=1);
+
 namespace PHPMD\TextUI;
 
+use CodeClimate\PHPMD\Runner;
+
 error_reporting(E_ERROR | E_PARSE | E_NOTICE);
 date_default_timezone_set('UTC');
 ini_set('memory_limit', -1);
 
 require_once __DIR__.'/vendor/autoload.php';
 require_once "JSONRenderer.php";
+require_once "Runner.php";
 
 use PHPMD\PHPMD;
 use PHPMD\RuleSetFactory;
 use PHPMD\Writer\StreamWriter;
 use PHPMD\Renderer\JSONRenderer;
 
-$config = json_decode(file_get_contents('/config.json'), true);
-
-$renderer = new JSONRenderer();
-$renderer->setWriter(new StreamWriter(STDOUT));
-
-$ruleSetFactory = new RuleSetFactory();
-
-$all_files = scandir_recursive("/code");
+$runner = new Runner();
 
-if ($config["exclude_paths"]) {
-  $files = array();
+// setup forking daemon
+$server = new \fork_daemon();
+$server->max_children_set(20);
+$server->max_work_per_child_set(50);
```

Comment:
Sure, `max_children_set` with the value of 20 means that the script with create up to 20 forks / processes, each processing up to 50 pieces of work (`max_work_per_child_set`).  The pieces of work in this case are the PHP files to scan.